payjoin-cli/src/app/config.rs

```diff
@@ -58,55 +58,53 @@ pub struct Config {
 }
 
 impl Config {
-    /// Version flags in order of precedence (newest to oldest)
-    const VERSION_FLAGS: &'static [(&'static str, u8)] = &[("bip77", 2), ("bip78", 1)];
-
     /// Check for multiple version flags and return the highest precedence version
-    fn determine_version(matches: &ArgMatches) -> Result<u8, ConfigError> {
+    fn determine_version(cli: &Cli) -> Result<u8, ConfigError> {
         let mut selected_version = None;
-        for &(flag, version) in Self::VERSION_FLAGS.iter() {
-            if matches.get_flag(flag) {
-                if selected_version.is_some() {
-                    return Err(ConfigError::Message(format!(
-                        "Multiple version flags specified. Please use only one of: {}",
-                        Self::VERSION_FLAGS
-                            .iter()
-                            .map(|(flag, _)| format!("--{flag}"))
-                            .collect::<Vec<_>>()
-                            .join(", ")
-                    )));
-                }
-                selected_version = Some(version);
+
+        // Check for BIP77 (v2)
+        if cli.flags.bip77.unwrap_or(false) {
+            selected_version = Some(2);
+        }
+
+        // Check for BIP78 (v1)
+        if cli.flags.bip78.unwrap_or(false) {
+            if selected_version.is_some() {
+                return Err(ConfigError::Message(
+                    "Multiple version flags specified. Please use only one of: --bip77, --bip78"
+                        .to_string(),
+                ));
             }
+            selected_version = Some(1);
         }
 
-        if let Some(version) = selected_version {
-            return Ok(version);
+        // If no version explicitly selected, use default based on available features
+        if selected_version.is_none() {
+            #[cfg(feature = "v2")]
+            return Ok(2);
+            #[cfg(all(feature = "v1", not(feature = "v2")))]
+            return Ok(1);
+            #[cfg(not(any(feature = "v1", feature = "v2")))]
+            return Err(ConfigError::Message(
+                "No valid version available - must compile with v1 or v2 feature".to_string(),
+            ));
         }
 
-        #[cfg(feature = "v2")]
-        return Ok(2);
-        #[cfg(all(feature = "v1", not(feature = "v2")))]
-        return Ok(1);
-
-        #[cfg(not(any(feature = "v1", feature = "v2")))]
-        return Err(ConfigError::Message(
-            "No valid version available - must compile with v1 or v2 feature".to_string(),
-        ));
+        Ok(selected_version.unwrap())
```

Comment:
Prefer not to use `.unwrap()` because if the code above changes then we could get a panic here.  Please short circuit instead and let the long circuit execute the implicit `is_none()` case

```rs
        if let Some(version) = selected_version {
            return Ok(version);
        }
        
        // If no version explicitly selected, use default based on available features
        #[cfg(feature = "v2")]
        return Ok(Version::TWO);
        #[cfg(all(feature = "v1", not(feature = "v2")))]
        return Ok(Version::ONE);
        #[cfg(not(any(feature = "v1", feature = "v2")))]
        return Err(ConfigError::Message(
            "No valid version available - must compile with v1 or v2 feature".to_string(),
        ));
```

Reply:
fixed